lib/puppet/transaction/report.rb

```diff
@@ -386,6 +394,89 @@ def self.default_format
 
   private
 
+  # Using the last applied values from our persistence store, and the current
+  # report data, iterate across all events finding any corrective_change that
+  # might exist and marking the event, resource and report appropriately.
+  def calculate_corrective_change
+    persistence = Puppet::Transaction::Persistence.new
+    persistence.load
+
+    # Prepare last transaction/new transaction data structures
+    last_resources = persistence.resources
+    new_resources = {}
+
+    resource_statuses.each do |name, status|
+      # Initialize data structure for resource
+      new_resources[name] ||= {}
+      params = new_resources[name]["parameters"] ||= {}
+
+      # Populate new hash with events mapped to params
+      param_to_event = {}
+      status.events.each do |ev|
```

Comment:
My implementation of event redaction does so as events are generated, so I think that my changes will strip out the information you're using before you can look at it. We'll have to sync up our two approaches.